### PR TITLE
[TTAHUB-2942] Fix selected goal count

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

After looking into this bug it seems there was much more broken on this page than we first thought.

The following logic is now in place:

- The 'Select all' check box is respected per page (IE its only selected if all goals on that PAGE are selected or all goals on each page are selected)
- We now track checkbox selection based on the actual number of goals selected (additional rolled up goal ids are looked up at print time) .
- The select goals on ALL pages link only shows if you have all goals selected on that specific page (and don't currently have all selected).
image
-  The pill of selected goals should show the number of goals selected on ALL page (so 10 on the first 2 on the second would show 12).
image
-  There are two values shown in the alert, the first is the number selected on this page the second is all that will be selected if you click the link.
image
- If no goals are selected we print all goals on that page ONLY.

## How to test

- Review the change.
- Test that the 'Select all' is respected per page.
- Check that we only show the select goals alert on all pages if all are selected on that page (and not already selected on all pages)
- Make sure we track selections globally (IE the first page might have all selected and the 3rd page only two goals).
- Test printing with all goals selected make sure rolled up goals are shown in the print.
- Test printing a page that has rolled up goals with nothing selected.
- Test printing with selected goal from various pages.
- Test printing with no goals selected.    

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2942

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Review by nvms — APPROVED** [review]
Nice!

File: src/services/recipient.js [hunk 1]
```diff
@@ -889,20 +889,23 @@ export async function getGoalsByActivityRecipient(
...
+  // For checkbox selection we only need the primary goal id.
+  const rolledUpGoalIds = r.goalRows.map((goal) => goal.id);
+
   if (limitNum) {
     return {
       count: r.goalRows.length,
       goalRows: r.goalRows.slice(offSetNum, offSetNum + limitNum),
       statuses,
-      allGoalIds,
+      allGoalIds: rolledUpGoalIds,
```
Comment (thewatermethod) [inline src/services/recipient.js:900]: Doesn't this mean that the places where we use allGoalIds on the frontend will now be missing goals? Print goals, I think, for example

Comment (AdamAdHocTeam) [reply src/services/recipient.js:900]: Looking at the history I think I added this for this exact purpose. I tested a case when I select a goal that is rolled up (ie has two goals) then clicked print. Both goals show up in the print preview but rolled up.

Comment (AdamAdHocTeam) [reply src/services/recipient.js:900]: image

**Review by thewatermethod — CHANGES_REQUESTED** [review]
https://github.com/HHS/Head-Start-TTADP/assets/3288586/d157f395-46a7-48a6-847f-c4264c6c8245

See video. With no selections made, "preview and print" excludes rolled up goals.

**Comment by AdamAdHocTeam** [comment]
> Screen.Recording.2024-05-31.at.4.05.19.PM.mov 
> See video. With no selections made, "preview and print" excludes rolled up goals.

I was able to reproduce this on staging, but I will take a look.

**Review by thewatermethod — APPROVED** [review]
Looks great to me, thanks for tackling this at a deeper level

**Comment by thewatermethod** [comment]
Updates look good

**Pull request merged**